gobblin-core/src/main/java/gobblin/GobblinMetricsRegistry.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin;
+
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.collect.MapMaker;
+
+
+public class GobblinMetricsRegistry {
```

Comment:
Talked with Yinan about this. Decided not to move to Guice for now (need to figure out how to do it correctly, or what's the best way of handling the singleton without guice)